crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1576,11 +1571,7 @@ impl<'db> TypeInferenceBuilder<'db> {
 
                 // Call into the context expression inference to validate that it evaluates
                 // to a valid context manager.
-                let context_expression_ty = if target.is_some() {
-                    self.infer_standalone_expression(&item.context_expr)
-                } else {
-                    self.infer_expression(&item.context_expr)
-                };
+                let context_expression_ty = self.infer_standalone_expression(&item.context_expr);
```

Comment:
I changed this because tests were panicking when using `infer_expression` in the else branch. Not sure I totally understand what happened here, but probably had to do with my changing the usage of `infer_context_expression` such that types are being inferred earlier?

Reply:
I suspect it's related to the added support for `self.x` as a with statement target, meaning we are now inferring context managers with attribute targets as standalone expressions, and that leads to double inference of the same expression(s) if we don't always go through `infer_standalone_expression`.

I think this change is correct, given that the semantic index builder always creates a standalone expression for every `with` statement.